main.rb

```diff
@@ -0,0 +1,87 @@
+# frozen_string_literal: true
+
+module Enumerable
+  def my_each
+    i = 0
+    while i < size
+      yield (self[i])
+      i += 1
+    end
+    self
+  end
+
+  def my_each_with_index
+    i = 0
+    while i < size
+      yield(self[i], i)
+      i += 1
+      self
+    end
+  end
+
+  def my_select
+    result = []
+    my_each { |item| result.push(item) if yield(item) }
+    result
+  end
+
+  def my_all?
+    my_each { |item| eturnr false if yield(item) == false }
```

Comment:
Always run your code before review....
`return false` not `eturnr false` :slightly_smiling_face: